tsl/src/bgw_policy/compression_api.c

```diff
@@ -9,51 +9,108 @@
 #define DEFAULT_MAX_RUNTIME                                                                        \
-	DatumGetIntervalP(DirectFunctionCall7(make_interval,                                           \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Float8GetDatum(0)))
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("0"), InvalidOid, -1))
+
 /* Right now, there is an infinite number of retries for compress_chunks jobs */
 #define DEFAULT_MAX_RETRIES -1
 /* Default retry period for reorder_jobs is currently 1 hour */
 #define DEFAULT_RETRY_PERIOD                                                                       \
-	DatumGetIntervalP(DirectFunctionCall7(make_interval,                                           \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(1),                                        \
-										  Int32GetDatum(0),                                        \
-										  Float8GetDatum(0)))
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("1 hour"), InvalidOid, -1))
+
+#define POLICY_COMPRESS_PROC_NAME "policy_compression"
+#define CONFIG_KEY_HYPERTABLE_ID "hypertable_id"
+#define CONFIG_KEY_OLDER_THAN "older_than"
+
+int32
+policy_compression_get_hypertable_id(Jsonb *config)
+{
+	bool found;
+	int32 hypertable_id = ts_jsonb_get_int32_field(config, CONFIG_KEY_HYPERTABLE_ID, &found);
+
+	if (!found)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("could not find hypertable_id in config for job")));
+
+	return hypertable_id;
+}
+
+int64
+policy_compression_get_older_than_int(Jsonb *config)
+{
+	bool found;
+	int32 hypertable_id = ts_jsonb_get_int64_field(config, CONFIG_KEY_OLDER_THAN, &found);
+
+	if (!found)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("could not find older_than in config for job")));
+
+	return hypertable_id;
+}
+
+Interval *
+policy_compression_get_older_than_interval(Jsonb *config)
+{
+	Interval *interval = ts_jsonb_get_interval_field(config, CONFIG_KEY_OLDER_THAN);
+
+	if (interval == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("could not find older_than in config for job")));
+
+	return interval;
+}
+
+static bool
+lag_equal(Oid partitioning_type, Jsonb *config, Oid lag_type, Datum lag_datum)
```

Comment:
I know what you mean by lag because you mentioned earlier. But it seems way too generic a term to use by itself without context or any comment. Was confused at first what this meant.

I would suggest at least calling it `compression_lag_equal`